packages/phone-number-privacy/signer/src/common/web3/contracts.ts

```diff
@@ -0,0 +1,197 @@
+    _getTransactionCount,
+    addresses,
+    (err: any) => {
+      throw err
+    },
+    Histograms.getRemainingQueryCountInstrumentation,
+    'getTransactionCount'
+  )
+}
+
+export async function getStableTokenBalance(
+  kit: ContractKit,
+  stableToken: StableToken,
+  logger: Logger,
+  ...addresses: string[]
+): Promise<BigNumber> {
+  const _getStableTokenBalance = (...params: string[]) =>
+    Promise.all(
+      params
+        .filter((address) => address !== NULL_ADDRESS)
+        .map((address) =>
+          retryAsyncWithBackOffAndTimeout(
+            async () => (await kit.contracts.getStableToken(stableToken)).balanceOf(address),
+            RETRY_COUNT,
+            [],
+            RETRY_DELAY_IN_MS,
+            undefined,
+            FULL_NODE_TIMEOUT_IN_MS
+          ).catch((err) => {
+            Counters.blockchainErrors.labels(Labels.read).inc()
+            throw err
+          })
+        )
+    ).then((values) => {
+      logger.trace(
+        { addresses, balances: values.map((bn) => bn.toString()) },
+        `Fetched ${stableToken} balances for addresses`
+      )
+      return values.reduce((a, b) => a.plus(b))
+    })
+  return meter(
+    _getStableTokenBalance,
+    addresses,
+    (err: any) => {
+      throw err
+    },
+    Histograms.getRemainingQueryCountInstrumentation,
+    'getStableTokenBalance'
+  )
+}
+
+export async function getCeloBalance(
+  kit: ContractKit,
+  logger: Logger,
+  ...addresses: string[]
+): Promise<BigNumber> {
+  const _getCeloBalance = (...params: string[]) =>
+    Promise.all(
+      params
+        .filter((address) => address !== NULL_ADDRESS)
+        .map((address) =>
+          retryAsyncWithBackOffAndTimeout(
+            async () => (await kit.contracts.getGoldToken()).balanceOf(address),
+            RETRY_COUNT,
+            [],
+            RETRY_DELAY_IN_MS,
+            undefined,
+            FULL_NODE_TIMEOUT_IN_MS
+          ).catch((err) => {
+            Counters.blockchainErrors.labels(Labels.read).inc()
+            throw err
+          })
+        )
+    ).then((values) => {
+      logger.trace(
+        { addresses, balances: values.map((bn) => bn.toString()) },
+        'Fetched celo balances for addresses'
+      )
+      return values.reduce((a, b) => a.plus(b))
+    })
+  return meter(
+    _getCeloBalance,
+    addresses,
+    (err: any) => {
+      throw err
+    },
+    Histograms.getRemainingQueryCountInstrumentation,
+    'getStableTokenBalance'
+  )
+}
+
+export async function getWalletAddress(
+  kit: ContractKit,
+  logger: Logger,
+  account: string
+): Promise<string> {
+  return meter(
+    retryAsyncWithBackOffAndTimeout,
+    [
+      async () => (await kit.contracts.getAccounts()).getWalletAddress(account),
+      RETRY_COUNT,
+      [],
+      RETRY_DELAY_IN_MS,
+      undefined,
+      FULL_NODE_TIMEOUT_IN_MS,
+    ],
+    (err: any) => {
+      logger.error({ error: err, account }, 'failed to get wallet address for account')
```

Comment:
I think the `error: err` part is failing in logging; it's displaying an empty {}. Looks like bunyan will automatically format the error and display everything properly if we instead use `err` as the field name (i.e. `logger.error({err, account})`). (Applies for all of the helper functions in this file

Reply:
actually wait I'm just gonna fix it in my PR since I'm touching this file anyways